app/services/gtfs_graph.rb

```diff
@@ -15,34 +15,61 @@ def initialize(filename, feed, feed_version)
     @onestop_id_to_entity = {}
   end
 
-  def create_change_osr(import_level=0)
-    raise ArgumentError.new('import_level must be 0, 1, or 2.') unless (0..2).include?(import_level)
-    log "Load GTFS"
+  def create_change_osr(import_level)
     @gtfs.load_graph
-    log "Load TL"
-    load_tl_stops
-    load_tl_routes
-    operators = load_tl_operators
-    routes = operators.map { |operator| operator.serves }.reduce(Set.new, :+)
-    stops = routes.map { |route| route.serves }.reduce(Set.new, :+)
-    log "Create changeset"
+    operators = Set.new
+    routes = Set.new
+    stops = Set.new
+
+    # Operators
+    @feed.operators_in_feed.each do |oif|
+      operator_gtfs = @gtfs.agency(oif.gtfs_agency_id)
+      next unless operator_gtfs
+      operator = find_by_onestop_id(oif.operator.onestop_id)
+      add_identifier(operator, 'o', operator_gtfs)
+      operators << operator
+      # Routes
+      @gtfs.children(operator_gtfs).each do |route_gtfs|
+        # Stops
+        route_stops = Set.new
+        route_stops_gtfs = @gtfs
+          .children(route_gtfs)
+          .map { |trip_gtfs| @gtfs.children(trip_gtfs) }
+          .reduce(Set.new, :+)
+        route_stops_gtfs.each do |stop_gtfs|
+          stop = find_by_entity(StopPlatform.from_gtfs(stop_gtfs))
+          stop.timezone ||= operator.timezone
+          add_identifier(stop, 's', stop_gtfs)
+          route_stops << stop
+          # Stations
+          stop_gtfs_parent = @gtfs.stop(stop_gtfs.parent_station)
+          if stop_gtfs_parent
+            station = find_by_entity(StopStation.from_gtfs(stop_gtfs_parent))
+            stop.parent_stop = station
+            add_identifier(station, 's', stop_gtfs_parent)
+            stops << station
+          end
+        end
+        stops |= route_stops
+        route = find_by_entity(Route.from_gtfs(route_gtfs, route_stops))
+        route.operator = operator
+        route.serves ||= Set.new
+        route.serves |= route_stops
+        add_identifier(route, 'r', route_gtfs)
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.